src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,30 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        LocalDate dateFromFormed = LocalDate.parse(dateFrom, FORMATTER);
+        LocalDate dateToFormed = LocalDate.parse(dateTo, FORMATTER);
+        StringBuilder builder = new StringBuilder();
+
+        for (String name: names) {
+            int salary = 0;
+            for (String info: data) {
+                String[] information = info.split(" ");
+                LocalDate dateFormed = LocalDate.parse(information[0], FORMATTER);
+                if (!dateFormed.isAfter(dateToFormed) && !dateFormed.isBefore(dateFromFormed)) {
+                    if (name.equals(information[1])) {
+                        salary += Integer.parseInt(information[2])
+                                * Integer.parseInt(information[3]);
+                    }
+                }
+            }
+            builder.append(System.lineSeparator()).append(name).append(" - ").append(salary);
+        }
+        return "Report for period " + dateFrom + " - " + dateTo + builder;
```

Comment:
The return statement concatenates the report header with the salary information using the `StringBuilder`. However, the `builder` starts with a line separator, which might lead to an unexpected format in the report. Consider adjusting the `StringBuilder` initialization or appending logic to ensure the format matches the expected output.